### PR TITLE
add basic lexer

### PR DESCRIPTION
**Pull request by suneettipirneni**

### CODE REVIEW

File: src/lex/token.rs [hunk 1]
```diff
@@ -11,14 +10,18 @@ pub enum Token {
 	Percent,
 	Caret,
 	Ampersand,
-	Space,
-	Newline,
-	String(String),
-	Tab,
+	Identifier(String),
+	NumberLiteral(String),
```
Comment (megatank58) [inline src/lex/token.rs]: maybe this could be stored as i64?

File: Cargo.toml [hunk 2]
```diff
@@ -6,3 +6,5 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+itertools = "0.10.5"
+utf8-chars = "2.0.2"
```
Comment (cheeze2000) [inline Cargo.toml]: please end this file with a newline

File: src/lex/token.rs [hunk 3]
```diff
@@ -11,14 +10,18 @@ pub enum Token {
 	Percent,
 	Caret,
 	Ampersand,
-	Space,
-	Newline,
-	String(String),
-	Tab,
+	Identifier(String),
```
Comment (cheeze2000) [inline src/lex/token.rs:13]: are you not gonna lex keywords?

File: src/lex/token.rs [hunk 4]
```diff
@@ -11,14 +10,18 @@ pub enum Token {
...
+	NumberLiteral(String),
+	StringLiteral(String),
 	End,
 	Semicolon,
 	LessThan,
 	GreaterThan,
+	Pipe,
+	LeftCurlyBrace,
+	RightCurlyBrace,
+	LeftParenthesis,
```
Comment (cheeze2000) [inline src/lex/token.rs]: maybe a shorter token name like "LParen" and "LBrace"

**Review by cheeze2000 — COMMENTED** [review]
nitpicks

File: src/lex/token.rs (hunk 3 above)
Comment (megatank58) [reply src/lex/token.rs:13]: won't keywords just be a part of the enum as values

**Review by cheeze2000 — APPROVED** [review]

**Pull request merged**